portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase

```diff
@@ -22,242 +22,127 @@ definition {
 		}
 	}
 
-	@description = "This is a use case for LRQA-40687. Add and Edit an OAuth2 application."
-	test AddAndEditApplication {
-		property app.server.types = "jboss,tcserver,tomcat,weblogic,websphere,wildfly";
-		property database.types = "db2,hypersonic,mariadb,mysql,oracle,postgresql,sqlserver,sybase";
-
-		task ("Add an OAuth2 application") {
-			var baseURL = PropsUtil.get("portal.url");
-
-			Navigator.openSpecificURL(url = "${baseURL}/group/control_panel/manage?p_p_id=com_liferay_oauth2_provider_web_internal_portlet_OAuth2AdminPortlet");
-
-			OAuth2.addApplication(
-				applicationName = "OAuth Application",
-				callbackURI = "http://www.liferay.com",
-				clientProfile = "Web Application",
-				websiteURL = "http://www.liferay.com");
-		}
-
-		task ("Input new values") {
-			var generatedClientId = selenium.getElementValue("OAuth2#ADMIN_INPUT_FORM_CLIENT_ID");
-			var generatedClientSecret = selenium.getElementValue("OAuth2#ADMIN_INPUT_FORM_CLIENT_SECRET");
-
-			OAuth2.inputApplicationValues(
-				description = "Registering a new application",
-				privacyPolicyURL = "http://www.liferay.com");
-
-			Navigator.gotoBack();
-		}
+	@description = "This is a use case for LPS-129398. Remember device with multiple auth without session."
+	@priority = "4"
+	test ApplicationCanBeAuthorizedWithoutUserSession {
+		property test.name.skip.portal.instance = "OAuth2#ApplicationCanBeAuthorizedWithoutUserSession";
 
-		task ("Assert correct application name, client Id and description") {
-			OAuth2.viewApplicationInTable(
-				applicationName = "OAuth Application",
-				clientId = "${generatedClientId}",
-				description = "Registering a new application");
-		}
+		task ("Add and edit 2 external applications") {
 
-		task ("Check OAuth application informations") {
-			OAuth2.viewApplication(
-				applicationName = "OAuth Application",
-				callbackURI = "http://www.liferay.com",
-				checkboxCheckList = "Authorization Code,Client Credentials,Refresh Token,Resource Owner Password Credentials",
-				clientId = "${generatedClientId}",
-				clientProfile = "Web Application",
-				clientSecret = "${generatedClientSecret}",
-				description = "Registering a new application",
-				privacyPolicyURL = "http://www.liferay.com",
-				websiteURL = "http://www.liferay.com");
-		}
+			// Test Setup
```

Comment:
Question for the future:
What do you think, shall we keep using these comments, or group setup steps into tasks?
@weller0246 @ferenconodiD ?

Reply:
I'm voting for tasks